protofsm/state_machine.go

```diff
@@ -346,17 +348,31 @@ func (s *StateMachine[Event, Env]) executeDaemonEvent(
 
 			// If a post-send event was specified, then we'll funnel
 			// that back into the main state machine now as well.
-			return fn.MapOptionZ(daemonEvent.PostSendEvent, func(event Event) error { //nolint:lll
-				return s.wg.Go(func(ctx context.Context) {
-					log.Debugf("FSM(%v): sending "+
-						"post-send event: %v",
-						s.cfg.Env.Name(),
-						lnutils.SpewLogClosure(event),
+			ok := fn.ElimOption(
```

Comment:
Why the switch to `ElimOption` here with the `func() bool { return true },` function? Can't this just be `MapOptionZ` with a boolean returned instead?

Reply:
Ah, right, makes sense :+1: